src/stemmabench/algorithms/stemma_NJ.py

```diff
@@ -0,0 +1,161 @@
+            edge_list=edges_list), text_list=list(self.manuscripts.keys()))
+        out.set_edges(edges_dict)
+        return out
+
+    def dist(self, distance: Callable) -> None:
+        """Builds the distance matix based on the provided distance function and sets the attribute _dist_matrix.
+
+        ### Args:
+            - distance (Callable): A function that takes as parameters 2 strings and that returns the distance between them.
+
+        ### Raises:
+            - RuntimeError: If method is called and the manuscripts attribute is not set.
+        """
+        if not self.manuscripts:
+            raise RuntimeError(
+                "The manuscripts attribute needs the be set before the distance can be calculated.")
+        self._dist_matrix = np.ndarray(
+            (len(self.manuscripts), len(self.manuscripts)))
+        # TODO: use map instead
+        for key, row in zip(sorted(self.manuscripts.keys()), range(len(self.manuscripts))):
+            for text, col in zip([self.manuscripts[k] for k in sorted(self.manuscripts.keys())], range(len(self.manuscripts))):
+                self._dist_matrix[row][col] = distance(
+                    self.manuscripts[key], text)
+
+    def _build_edges(self) -> Tuple[Dict[str, float], list[list[str]]]:
+        """Builds list of edges as well as the associated dictionayr containing the edge distances.
+
+        ### Returns:
+            - dict: The dictionary with edges as keys and distences as values.
+            - list: List of edges.
+        """
+        if not isinstance(self._dist_matrix, np.ndarray):
+            raise RuntimeError(
+                "Can not call _build_edges if _dist_matrix has not been initialized.")
+        temp_dist_matrix = self._dist_matrix.copy()
+        labels = sorted(list(self.manuscripts.keys()))
+        edges_labels = []
+        edges_distance = []
+        while temp_dist_matrix.shape[0] > 2:
+            temp_dist_matrix, labels, df, f_lab, dg, g_lab = self._aglo(
+                temp_dist_matrix, labels)
+            edges_labels.append([labels[len(labels)-1], f_lab])
+            edges_distance.append(df)
+            edges_labels.append([labels[len(labels)-1], g_lab])
+            edges_distance.append(dg)
+        edges_distance.append(temp_dist_matrix[0, 1])
+        edges_labels.append([labels[0], labels[1]])
+        edges_dict_labels = [l[0] + "," + l[1] for l in edges_labels]
+        return {edges_dict_labels[i]: edges_distance[i] for i in range(len(edges_dict_labels))}, edges_labels
+
+    def _aglo(self, dist_mat: np.ndarray, labels: list) -> Tuple[np.ndarray, list[str], float, str, float, str]:
+        """Performs one step in the distance matrix aglomeration process and returns all information needed for Neighbour Joining.
+           Does not work for matrix 2*2. 
+
+        ### Args:
+            - dist_mat (np.ndarray): A !!!insert name here!!! distance matrix to be aglomerated by one step.
```

Comment:
```suggestion
            - dist_mat (np.ndarray): A !!!insert name here!!! distance matrix to be agglomerated by one step.
```

Why insert name here ?

Reply:
Corrected